sale_order_type/views/sale_order_view.xml

```diff
@@ -6,7 +6,12 @@
         <field name="inherit_id" ref="sale.view_order_form"/>
         <field name="arch" type="xml">
             <field name="currency_id" position="after">
-                <field name="type_id" required="1"/>
+                <label for="type_id"/>
+                <div name="order_type" class="o_row">
+                    <field name="type_id" required="1"/>
+                    <button type="object" name="match_order_type"
+                            string="Find by rule" class="oe_inline" icon="fa-gears"/>
```

Comment:
This should be also done for invoices

Reply:
Done, along with some refactoring, in https://github.com/OCA/sale-workflow/pull/716/commits/148e7b60af81508b48788a45021d44f6d8e15ca3